lib/teleterm/gateway/gateway.go

```diff
@@ -139,6 +160,12 @@ type Gateway struct {
 	localProxy *alpn.LocalProxy
 	// closeContext and closeCancel are used to signal to any waiting goroutines
 	// that the local proxy is now closed and to release any resources.
-	closeContext context.Context
-	closeCancel  context.CancelFunc
+	closeContext       context.Context
+	closeCancel        context.CancelFunc
+	cliCommandProvider CLICommandProvider
```

Comment:
I refactored the code a bit, the summary of the changes is:

* `Gateway` now has `SetTargetSubresourceName` instead of this method being defined on `Cluster`.
* `cliCommandProvider` has been moved from `Cluster` to `Gateway`.
* `Gateway` has been modified so that it accepts `cliCommandProvider` during
  its creation.
* Later it can call `cliCommandProvider` to get the CLI command when needed.
* Setting the CLI command is no longer done outside of `Gateway`.